gammapy/irf/energy_dispersion.py

```diff
@@ -718,17 +730,23 @@ def __init__(
     ):
         if interp_kwargs is None:
             interp_kwargs = self.default_interp_kwargs
-        axes = [
-            BinnedDataAxis(
-                e_true_lo, e_true_hi, interpolation_mode="log", name="e_true"
-            ),
-            BinnedDataAxis(
-                migra_lo, migra_hi, interpolation_mode="linear", name="migra"
-            ),
-            BinnedDataAxis(
-                offset_lo, offset_hi, interpolation_mode="linear", name="offset"
-            ),
-        ]
+
+
+        e_true_edges = np.append(e_true_lo, e_true_hi[-1]).value * e_true_lo.unit
```

Comment:
Maybe all these lines could be replaced by a simple helper function. This could allow to add a  check to ensure that lo and hi values are contiguous.

Reply:
See my comment above.